charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -111,13 +124,45 @@ Get otel memory_limiter ballast_size_mib value based on 40% of resources.memory.
 {{- div (mul (include "opentelemetry-collector.convertMemToMib" .) 40) 100 }}
 {{- end -}}
 
-{{- define "opentelemetry-collector.daemonset.containerLogsConfig" -}}
-{{- if .Values.containerLogs.enabled }}
+{{- define "opentelemetry-collector.applyHostMetricsConfig" -}}
+{{- $config := mustMergeOverwrite (include "opentelemetry-collector.hostMetricsConfig" .Values | fromYaml) .config }}
+{{- $newList := dict "service" (dict "pipelines" (dict "metrics" (dict "receivers" (append .config.service.pipelines.metrics.receivers "hostmetrics" | uniq))))}}
+{{- $config := mustMergeOverwrite $config $newList  }}
```

Comment:
I believe we can use `set` http://masterminds.github.io/sprig/dicts.html with updated list of receivers instead of `mustMergeOverwrite`

Reply:
Unfortunately I don't think `set` knows how to handle nested objects.  retrying to set a nested object like `"service.pipelines.logs.receivers"` resulted in a new root-level field called `"service.pipelines.logs.receivers"`

Reply:
Done.